include/dt-bindings/pinctrl/stm32-pinctrl-common.h

```diff
@@ -7,6 +7,61 @@
 #ifndef ZEPHYR_STM32_PINCTRL_COMMON_H_
 #define ZEPHYR_STM32_PINCTRL_COMMON_H_
 
+/* Adapted from Linux: include/dt-bindings/pinctrl/stm32-pinfunc.h */
+
+/**
+ * @brief Pin modes
+ */
+
+#define AF0	0x0
+#define AF1	0x1
+#define AF2	0x2
+#define AF3	0x3
+#define AF4	0x4
+#define AF5	0x5
+#define AF6	0x6
+#define AF7	0x7
+#define AF8	0x8
+#define AF9	0x9
+#define AF10	0xa
+#define AF11	0xb
+#define AF12	0xc
+#define AF13	0xd
+#define AF14	0xe
+#define AF15	0xf
+#define ANALOG	0x10
+#define GPIO    0x11 /* Reserved for future use */
+
+/**
+ * @brief numerical IDs for IO ports
+ */
+
+#define	STM32_DT_PORT_a 0	/* IO port A */
+#define	STM32_DT_PORT_b 0x10	/* .. */
+#define	STM32_DT_PORT_c 0x20
+#define	STM32_DT_PORT_d 0x30
+#define	STM32_DT_PORT_e 0x40
+#define	STM32_DT_PORT_f 0x50
+#define	STM32_DT_PORT_g 0x60
+#define	STM32_DT_PORT_h 0x70
+#define	STM32_DT_PORT_i 0x80
+#define	STM32_DT_PORT_j 0x90
+#define	STM32_DT_PORT_k 0xA0	/* IO port K */
+
+/**
+ * @brief Macro to generate pinmux int using port, pin number and mode arguments
+ * This is a variation to Linux STM32_PINMUX macro which is usable in
+ * preprocessor macros.
+ */
+
+#define STM32_PINMUX_Z(port, pin, mode) ((((port + pin)) << 8) | (mode))
```

Comment:
why adding directly port and pin ? Why  not concatenating them like that:
```suggestion
#define STM32_PINMUX_Z(port, pin, mode) ((port) << 12 | (pin) << 8) | (mode))
```

define should be adapted :
#define	STM32_DT_PORT_b 0x1

Then shift value (8 and 12) could be also #define  resused in both STM32_PINMUX_Z , STM32_DT_PINMUX_PORT and STM32_DT_PINMUX_LINE)

Reply:
I actually stayed intentionally close to the equivalent linux macro.